src/kubectlUtils.ts

```diff
@@ -0,0 +1,113 @@
+import * as vscode from "vscode";
+import { Kubectl } from "./kubectl";
+import { kubeChannel } from "./kubeChannel";
+
+export interface Cluster {
+    readonly name: string;
+    readonly context: string;
+    readonly user: string;
+    readonly active: boolean;
+}
+
+export interface Namespace {
+    readonly name: string;
+    readonly active: boolean;
+}
+
+async function getKubeconfig(kubectl: Kubectl): Promise<any> {
+    const shellResult = await kubectl.invokeAsync("config view -o json");
+    if (shellResult.code !== 0) {
+        vscode.window.showErrorMessage(shellResult.stderr);
+        return null;
+    }
+    return JSON.parse(shellResult.stdout);
+}
+
+export async function getClusters(kubectl: Kubectl): Promise<Cluster[]> {
+    const kubectlConfig = await getKubeconfig(kubectl);
+    if (!kubectlConfig) {
+        return [];
+    }
+    const currentContext = kubectlConfig["current-context"];
+    const contexts = kubectlConfig.contexts;
+    return contexts.map((c) => {
+        return {
+            name: c.context.cluster,
+            context: c.name,
+            user: c.context.user,
+            active: c.name === currentContext
+        };
+    });
+}
+
+export async function deleteCluster(kubectl: Kubectl, cluster: Cluster): Promise<boolean> {
+    const deleteClusterResult = await kubectl.invokeAsyncWithProgress(`config delete-cluster ${cluster.name}`, "Deleting cluster...");
+    if (deleteClusterResult.code !== 0) {
+        kubeChannel.showOutput(`Failed to delete the specified cluster ${cluster.name} from the kubeconfig: ${deleteClusterResult.stderr}`, `Delete-${cluster.name}`);
+        vscode.window.showErrorMessage("Failed. See the output window for more details.");

```

Comment:
I think the first sentence might strip _too much_ context.  I'd suggest at minimum 'Delete failed.' and perhaps 'Delete ${cluster.name} failed.'  We're not completely strapped for space in the header - we just can't always handle what can be lengthy error messages.